ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -125,6 +142,16 @@ static void pdo_pgsql_fetch_error_func(pdo_dbh_t *dbh, pdo_stmt_t *stmt, zval *i
 }
 /* }}} */
 
+static void pdo_pgsql_cleanup_notice_callback(pdo_pgsql_db_handle *H) /* {{{ */
+{
+	if (H->notice_callback) {
```

Comment:
The code seems strange here, you should free the fcc using `zend_fcc_dtor` and not manualy handle the fci.